From 4abfc9a93e3fa96707147c510c6ae9aaa7a3a434 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 02 May 2019 05:05:17 +0300
Subject: [PATCH] [SW-0] pcie: do per platform handling

In ONIX, and PLDM it is only an empty stub which returns that no PCIe
init is needed.

Change-Id: I26b1cecb5d6e8ebc5bbd7820fa02d6ccd77e45aa
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 9cb7f88..638c6d0 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -115,3 +115,9 @@
 	return pll_num;
 }
 
+u32 hl_skip_pcie_init(void)
+{
+	/* Use DumpSel[11] to skip PCIe initialization */
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
+}
+
diff --git a/gaudi/common.c b/gaudi/common.c
index f7a61c9..ba10af4 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -44,12 +44,6 @@
 	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
 }
 
-u32 hl_skip_pcie_init(void)
-{
-	/* Use DumpSel[11] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
-}
-
 u64 hl_pll_get_cpu_pll_base_off(void)
 {
 	return mmPSOC_CPU_PLL_BASE;
diff --git a/gaudi/onix.c b/gaudi/onix.c
index 85badcc..c6a57ec 100644
--- a/gaudi/onix.c
+++ b/gaudi/onix.c
@@ -12,3 +12,8 @@
 {
 	return 0;
 }
+
+u32 hl_skip_pcie_init(void)
+{
+	return 1;
+}
diff --git a/gaudi/pldm.c b/gaudi/pldm.c
index 85badcc..c6a57ec 100644
--- a/gaudi/pldm.c
+++ b/gaudi/pldm.c
@@ -12,3 +12,8 @@
 {
 	return 0;
 }
+
+u32 hl_skip_pcie_init(void)
+{
+	return 1;
+}
diff --git a/goya/asic.c b/goya/asic.c
index 8221909..61e5a5b 100644
--- a/goya/asic.c
+++ b/goya/asic.c
@@ -7,6 +7,8 @@
 
 #include "common.h"
 
+#include <psoc_global_conf_regs.h>
+
 #include <preboot.h>
 #include <pll.h>
 
@@ -35,3 +37,9 @@
 	return pll_num;
 }
 
+u32 hl_skip_pcie_init(void)
+{
+	/* Use DumpSel[8] to skip PCIe initialization */
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(21);
+}
+
diff --git a/goya/common.c b/goya/common.c
index b293a3b..6abe158 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -171,12 +171,6 @@
 	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
 }
 
-u32 hl_skip_pcie_init(void)
-{
-	/* Use DumpSel[8] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(21);
-}
-
 u64 hl_pll_get_cpu_pll_base_off(void)
 {
 	return mmCPU_PLL_BASE;
diff --git a/goya/onix.c b/goya/onix.c
index 85badcc..c6a57ec 100644
--- a/goya/onix.c
+++ b/goya/onix.c
@@ -12,3 +12,8 @@
 {
 	return 0;
 }
+
+u32 hl_skip_pcie_init(void)
+{
+	return 1;
+}
diff --git a/goya/pldm.c b/goya/pldm.c
index 85badcc..c6a57ec 100644
--- a/goya/pldm.c
+++ b/goya/pldm.c
@@ -12,3 +12,8 @@
 {
 	return 0;
 }
+
+u32 hl_skip_pcie_init(void)
+{
+	return 1;
+}
